### PR TITLE
ScatterView: fix for OpenmpTarget remove inheritance from reducers

### PR DESCRIPTION
**Pull request by crtrott**
Inheriting from Reducers broke OpenMPTarget, but it also means
every scatter element has unnecessarily a view member.
IMO there is also not code complexity gain from inheriting, rather
the opposite.

### CODE REVIEW

File: containers/src/Kokkos_ScatterView.hpp [hunk 1]
```diff
@@ -199,35 +211,30 @@ struct ScatterValue;
...
+                    Kokkos::Experimental::ScatterNonAtomic> {
+  ValueType& value;
+
  public:
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ValueType& value_in)
-      : Sum<ValueType, DeviceType>(value_in) {}
+      : value(value_in) {}
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ScatterValue&& other)
-      : Sum<ValueType, DeviceType>(other.reference()) {}
+      : value(other.value) {}
```
Comment (dalg24) [inline containers/src/Kokkos_ScatterView.hpp:221]: Why do we provide a move constructor in the first place?

File: containers/src/Kokkos_ScatterView.hpp [hunk 2]
```diff
@@ -199,35 +211,30 @@ struct ScatterValue;
...
+  KOKKOS_FORCEINLINE_FUNCTION void operator++(int) { update(1); }
   KOKKOS_FORCEINLINE_FUNCTION void operator-=(ValueType const& rhs) {
-    this->join(this->reference(), ValueType(-rhs));
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator--() {
-    this->join(this->reference(), ValueType(-1));
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator--(int) {
-    this->join(this->reference(), ValueType(-1));
+    update(ValueType(-rhs));
```
Comment (dalg24) [inline containers/src/Kokkos_ScatterView.hpp:228]: Do we actually need that cast to ValueType?

File: containers/src/Kokkos_ScatterView.hpp [hunk 3]
```diff
@@ -350,9 +362,11 @@ struct ScatterValue<ValueType, Kokkos::Experimental::ScatterProd, DeviceType,
   }
 
   KOKKOS_FORCEINLINE_FUNCTION void update(ValueType const& rhs) {
-    this->join(this->reference(), rhs);
+    value *= rhs;
```
Comment (dalg24) [inline containers/src/Kokkos_ScatterView.hpp:365]: Does this guy not need to be atomic?

File: containers/src/Kokkos_ScatterView.hpp (hunk 1 above)
Comment (crtrott) [reply containers/src/Kokkos_ScatterView.hpp:221]: I didn't want to mess with it here. But yeah I didn't know either really.

File: containers/src/Kokkos_ScatterView.hpp (hunk 2 above)
Comment (crtrott) [reply containers/src/Kokkos_ScatterView.hpp:228]: probably not but I didn't want to mess with it either.

**Comment by dalg24** [comment]
Retest this please

File: containers/src/Kokkos_ScatterView.hpp (hunk 3 above)
Comment (dalg24) [reply containers/src/Kokkos_ScatterView.hpp:365]: Ping

Comment (crtrott) [reply containers/src/Kokkos_ScatterView.hpp:365]: yeah

File: containers/src/Kokkos_ScatterView.hpp (hunk 1 above)
Comment (dhollman) [reply containers/src/Kokkos_ScatterView.hpp:221]: So as to disable the copy constructor?

Comment (dalg24) [reply containers/src/Kokkos_ScatterView.hpp:221]: With the introduction of the `value` data member that is a reference, the type is definitely not copyable nor assignable now :/

File: containers/src/Kokkos_ScatterView.hpp [hunk 4]
```diff
@@ -209,6 +209,11 @@ struct DefaultContribution<Kokkos::Experimental::HIP,
 template <typename ValueType, int Op, typename DeviceType, int contribution>
 struct ScatterValue;
 
+// FIXME All these scatter values need overhaul: 
+//   - like should they be copyable at all?
+//   - what is the internal handle type
+//   - remove join
+//   - constistenly use the update function in operators
```
Comment (dalg24) [inline containers/src/Kokkos_ScatterView.hpp:216]:
```suggestion
//   - consistently use the update function in operators
```

**Pull request merged**